### PR TITLE
tests/cri-containerd: split between cri-containerd and crictl suites

### LINKED ISSUE #8160: tests: split the cri-containerd tests
The cri-containerd suite of tests (tests/integration/cri-containerd) executes two types of tests:

1. The containerd's cri-integration suite (https://github.com/containerd/containerd/blob/main/docs/cri/testing.md#cri-integration-test)
2. Custom shell script tests within kata's code base, using crictl/cri/containerd

The tests (1) are written in Golang and external to Kata's source code, while (2) are shell-based tests that leverage crictl (https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md) to check kata's specific implementation (e.g. upgrade of container's memory w/ virtio-mem). Those are really different suite of tests which come to use the same containerd's CRI plug-in interface. 

So what I proposed here is to split those in two suites of tests. Some outcomes are:

* ease the interpretation of the logs as currently a developer not versed on kata's tests won't even notice there are different tests running. Allow to run in different CI jobs.
* can serve as a home for crictl-based tests from CCv0 to the main branch

Other than split the suites, I propose to convert the crictl tests from pure shell scripts to bats (https://bats-core.readthedocs.io).

### PR DESCRIPTION
**Pull request by wainersm**
This split the cri-containerd in two suites of tests:

* cri-containerd:  keep running containerd's [cri-integration tests](https://github.com/containerd/containerd/blob/main/docs/cri/testing.md#cri-integration-test)
* crictl: new home for the crictl-based tests 

Then the crictl tests were convert to use the bats framework. I didn't  want to improve the tests, depict many opportunities, so I tried to keep the original code and logic intact as much as possible. In the future, when we start pulling tests from CCv0 I think it will be the right time for improvements.

A github workflow was created to run the crictl suite of tests. The workflow won't fail due to failures on tests, because I will be able to fully test the tests on CI after the workflow is merged (right?) and then I don't want a broken workflow blocking unrelated pull requests.

Fixes #8160

### CODE REVIEW

**Comment by wainersm** [comment]
Hi @fidencio !

> @wainersm, I don't understand [9c1a213](https://github.com/kata-containers/kata-containers/commit/9c1a2132f31c631002b788b5c162cf89b250a179)
> 
> What exactly are you trying to gain with that? integration/cri-containerd should be clear that this is cri-containerd integration tests ... making it integration/cri-containerd-integration is redundant.

Once upon a time PR #8061, that @beraldoleal asked me to help understand why some CI jobs failed. I took look at https://github.com/kata-containers/kata-containers/actions/runs/6356069723/job/17266102296?pr=8061 first, when I realized that our `cri-containerd` suite of tests is actually running two types of tests:

 1. The containerd's cri-integration suite. External to kata's code base, writing in Golang...etc
 2. Custom shell script tests within kata's code base, using crictl/cri/containerd

So  my goal is to split those two types of tests in different suites to easy the interpretation of the logs when the CI fail. The tests 2) (`cri-containerd`) could even growth in the future with the addition of https://github.com/kata-containers/tests/tree/CCv0/integration/containerd/confidential tests.

Let me know if I'm missing something and/or this is an unwanted/worthless split.

**Comment by fidencio** [comment]
Aha, okay, it makes sense.

I'd just not name it as cri-containerd-integration :-)

**Comment by wainersm** [comment]
> Aha, okay, it makes sense.
> 
> I'd just not name it as cri-containerd-integration :-)

hmm.. what do you suggest? Keep containerd's cri-integration tests called `cri-containerd` and naming our custom tests something else (e.g. `crictl-containerd`) ? Naming is an art my friend...

**Comment by wainersm** [comment]
**run-cri-containerd (lts, qemu)** is also failing locally and I still don´t know why.

**Comment by wainersm** [comment]
> **run-cri-containerd (lts, qemu)** is also failing locally and I still don´t know why.

Fixed. Added an instruction to pull the pause image before creating the pod.

**Comment by wainersm** [comment]
Re-done my initial PR to move the crictl tests to its own suite (guess what, named `crictl`). So the cri-containerd suite is kept running the containerd's cri-integration tests.

Once CI passing I should be working on:
* convert crictl to use bats
* create a new CI job to run crictl only

**Comment by wainersm** [comment]
@fidencio this PR is ready to review. The three commits that convert to bats I should squash after the reviews (keeping them separated to ease the reviewer's life).
                                                                     / ---- \
                                                                    / ------ \
@stevenhorsman tentative to build a  |  []  |  (house) for the crictl tests that we will (or not) be pulling to main...

**Comment by wainersm** [comment]
> @fidencio this PR is ready to review. The three commits that convert to bats I should squash after the reviews (keeping them separated to ease the reviewer's life). / ---- / ------ @stevenhorsman tentative to build a | [] | (house) for the crictl tests that we will (or not) be pulling to main...

yeah, my tentative to build a house in ascii didn't work out....

**Comment by stevenhorsman** [comment]
/ ---- \
    / ------ \
      | [] |

File: tests/integration/crictl/lib.sh
Comment (stevenhorsman) [inline tests/integration/crictl/lib.sh:1]: Do we need a copyright statement here (presumably copied from the source of this code)?

**Review by stevenhorsman — APPROVED** [review]
Looks good to me, one minor comment...

**Comment by wainersm** [comment]
Rebase this to main, added a header to `tests/integration/crictl/lib.sh` and squashed the commits that convert tests to bats.

**Comment by stevenhorsman** [comment]
@wainersm - Do we still want to get this in? I think it needs a rebase and re-test and then we can have another attempt?

**Pull request closed without merging**